### PR TITLE
Share images from other apps directly to Openbook

### PR DESCRIPTION
**Pull request by 999eagle**
This implements image sharing and fixes issue #166. Currently still a WIP as the actual post creation isn't yet implemented.

### CODE REVIEW

**Comment by 999eagle** [comment]
Okay, latest two commits finished image sharing from other apps in Android. Openbook now shows up in the system share menu in other apps and when receiving an image, it will open the new post modal with the shared image already attached. Tested to work in virtual device and on my Android 7.1.2 phone.

**Comment by lifenautjoe** [comment]
Nice! Only thing left is to make sure the shared image goes through the same checks than a picked one. As the ratio doesn't matter, just checking that the file size is not too big.

We can do this as soon the image is receive by the `receiveShare` function with the `ValidationsService`, `isImageAllowedSize` with type `Post` and show a toast if too large in the same way it's done in the `createPost` widget (also not opening the modal).

In regards to naming, for the argument in the post modal, can we change it to just `image` instead of `sharedImage`? To make it clear that it can be really any image. 

And `receiveShare` to `onShare`, `enableReceiving` to `enableSharing` and finally `disableReceiving` to `disableSharing`?

Overall, awesome contribution @999eagle !

**Comment by lifenautjoe** [comment]
I've found this answer on Stackoverflow which seems to contain code for doing this on iOS https://stackoverflow.com/a/50007287/2608145

**Comment by 999eagle** [comment]
The answer you linked is only for sharing content from the flutter app to other apps, sadly not for receiving such a share.

**Comment by lifenautjoe** [comment]
Argh, alright. Confused over the word share in this context, it implies no direction.

Thanks for the changes!

**Comment by 999eagle** [comment]
Okay I've changed the names and added image size validation.

File: lib/pages/home/home.dart [hunk 1]
```diff
@@ -126,10 +130,16 @@ class OBHomePageState extends ReceiveShareState<OBHomePage>
...
   @override
-  void onShare(Share share) {
+  void onShare(Share share) async {
     debugPrint("received share to file " + share.path);
-    _modalService.openCreatePost(
-        context: context, image: File.fromUri(Uri.parse(share.path)));
+    var image = File.fromUri(Uri.parse(share.path));
+    if (!await _validationService.isImageAllowedSize(image, OBImageType.post)) {
+      int limit = _validationService.getAllowedImageSize(OBImageType.post);
+      _toastService.error(
```
Comment (lifenautjoe) [inline lib/pages/home/home.dart]: I think the int limit has to be divided for it to be MBs, so  `limit = limit ~/ 1048576;` 🙈

Comment (999eagle) [reply lib/pages/home/home.dart]: Oh.. Yeah, copied that from the photo picker, but seems like I missed that...

**Comment by lifenautjoe** [comment]
Looks good Sophie! Will test some more and should be merged soon :-)

**Pull request merged**